ast/src/call.rs

```diff
@@ -33,8 +33,11 @@ impl SourceSegmentHolder for Call<'_> {
 #[segment_holder]
 #[derive(Debug, Clone, PartialEq, DebugPls)]
 pub struct ProgrammaticCall<'a> {
-    /// The name of the function to call.
-    pub name: &'a str,
+    /// The function to call.
+    ///
+    /// This may or may not be a string literal, as it may a call on a variable, or a call on the
+    /// result of another call.
+    pub expr: Box<Expr<'a>>,
```

Comment:
You should rollback this to the previous `name: &'a str`, i see you made this to allow expressions like `$a(...)`, `{}(...)` etc to be represented by PFCs but this kind of expression should be represented by `MethodCall` as those are method calls